pkg/controller/atlasproject/project.go

```diff
@@ -168,5 +168,21 @@ func (r *AtlasProjectReconciler) hasDependencies(ctx *workflow.Context, project
 		return false, err
 	}
 
-	return len(streamInstances.Items) > 0, nil
+	if len(streamInstances.Items) > 0 {
+		return true, nil
+	}
+
+	privateEndpoints := &akov2.AtlasPrivateEndpointList{}
+	listOps = &client.ListOptions{
+		FieldSelector: fields.OneTermEqualSelector(
+			indexer.AtlasPrivateEndpointByProjectIndex,
+			client.ObjectKeyFromObject(project).String(),
+		),
+	}
+	err = r.Client.List(ctx.Context, privateEndpoints, listOps)
+	if err != nil {
+		return false, err
+	}
+
+	return len(privateEndpoints.Items) > 0, nil
 }
```

Comment:
So... we were ignoring the private endpoints as dependencies before?

Reply:
no, because it was a subresource, and when it is a sub-resource, it's done in a different place.